packages/fxa-auth-server/lib/payments/iap/apple-app-store/subscriptions.ts

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { IapExtraStripeInfo } from 'fxa-shared/dto/auth/payments/iap-subscription';
+import { MozillaSubscriptionTypes } from 'fxa-shared/subscriptions/types';
+import Container from 'typedi';
+
+import { internalValidationError } from '../../../../lib/error';
+import { AppConfig } from '../../../types';
+import { StripeHelper } from '../../stripe';
+import { SubscriptionsService } from '../types';
+import { AppleIAP } from './apple-iap';
+import { AppStoreSubscriptionPurchase } from './subscription-purchase';
+
+export class AppStoreSubscriptions
+  implements
+    SubscriptionsService<AppStoreSubscriptionPurchase & IapExtraStripeInfo>
+{
+  private appleIap?: AppleIAP;
+  private stripeHelper?: StripeHelper;
+
+  constructor() {
+    const config = Container.get(AppConfig);
+    if (!config.subscriptions.enabled) {
+      throw internalValidationError(
+        'AppStoreSubscriptions',
+        {},
+        new Error(
+          'Trying to new up AppStoreSubscriptions while subscriptions are disabled.  Check your dependency graph.'
```

Comment:
What does 'new up' mean? Did you mean that this shouldn't be instantiated?

Reply:
Yeah; this was CCP from a similar error message in for the Google Play analog, `PlaySubscriptions`. I also paused here, but I thought it was kind of amusing. I can change the error message in both cases to use "instantiate".